### PR TITLE
Boiler work

### PR DESCRIPTION
**Pull request by ExpensiveKoala**
This PR fixes the boiler and makes it functional. It also rewrites the TileEntity to use IItemHandler.

### CODE REVIEW

File: src/api/java/eiteam/esteemedinnovation/api/util/FluidHelper.java [hunk 1]
```diff
@@ -237,6 +237,9 @@ public static BlockPos findSourceBlockPos(World world, BlockPos start, Fluid flu
      * @return An {@link IFluidHandler} for the Tile and direction. If it uses the deprecated API, returns a new wrapper.
      */
     public static IFluidHandler getFluidHandler(TileEntity tile, EnumFacing dir) {
+        if(tile == null) {
```
Comment (elifoster) [inline src/api/java/eiteam/esteemedinnovation/api/util/FluidHelper.java]:
```suggestion
        if (tile == null) {
```

File: src/main/java/eiteam/esteemedinnovation/boiler/ContainerBoiler.java [hunk 2]
```diff
@@ -111,8 +112,8 @@ public void updateProgressBar(int id, int data) {
     }
 
     @Override
-    public boolean canInteractWith(@Nonnull EntityPlayer par1EntityPlayer) {
-        return tileEntity.isUsableByPlayer(par1EntityPlayer);
+    public boolean canInteractWith(EntityPlayer playerIn) {
+        return true;
```
Comment (elifoster) [inline src/main/java/eiteam/esteemedinnovation/boiler/ContainerBoiler.java:116]: Why was this changed?

File: src/main/java/eiteam/esteemedinnovation/boiler/ContainerBoiler.java [hunk 3]
```diff
@@ -111,8 +112,8 @@ public void updateProgressBar(int id, int data) {
     }
 
     @Override
-    public boolean canInteractWith(@Nonnull EntityPlayer par1EntityPlayer) {
```
Comment (elifoster) [inline src/main/java/eiteam/esteemedinnovation/boiler/ContainerBoiler.java:114]: Is this no longer `@Nonnull`?

File: src/main/java/eiteam/esteemedinnovation/boiler/TileEntityBoiler.java [hunk 4]
```diff
@@ -68,48 +77,51 @@ public static int getItemBurnTime(@Nonnull ItemStack stack) {
...
         Item item = stack.getItem();
-
+        int burnTime = ForgeEventFactory.getItemBurnTime(stack);
+        if (burnTime >= 0) {
+            return burnTime;
+        }
         if (item instanceof ItemBlock && Block.getBlockFromItem(item) != Blocks.AIR) {
             Block block = Block.getBlockFromItem(item);
 
             if (OreDictHelper.slabWoods.contains(item)) {
```
Comment (elifoster) [inline src/main/java/eiteam/esteemedinnovation/boiler/TileEntityBoiler.java:87]: Regarding these changes `return` -> `burnTime =`, I'm curious your rationale. It seems that it would be faster to return quick than to check a bunch of additional conditionals, no?

File: src/main/java/eiteam/esteemedinnovation/boiler/TileEntityBoiler.java [hunk 5]
```diff
@@ -195,29 +205,33 @@ public void safeUpdate() {
         }
         */
 
-        ItemStack stackInInput = getStackInSlot(1);
-        if (FluidHelper.itemStackIsWaterContainer(stackInInput)) {
-            ItemStack drainedItemStack = FluidHelper.fillTankFromItem(stackInInput, myTank, true);
-            setInventorySlotContents(1, drainedItemStack);
+        ItemStack stackInInput = inventory.getStackInSlot(1);
+        if (!stackInInput.isEmpty()) {
```
Comment (elifoster) [inline src/main/java/eiteam/esteemedinnovation/boiler/TileEntityBoiler.java]: This line is unnecessary, `FluidHelper.itemStackIsWaterContainer(ItemStack)` checks if the provided `ItemStack` is empty.

File: src/main/java/eiteam/esteemedinnovation/boiler/TileEntityBoiler.java [hunk 6]
```diff
@@ -195,29 +205,33 @@ public void safeUpdate() {
...
+                inventory.setStackInSlot(1, drainedItemStack);
+            }
         }
 
         boolean isBurnTimeGreaterThanZero = burnTime > 0;
         if (burnTime > 0) {
             //maxThisTick = Math.min(burnTime, 10);
             burnTime -= 1; //maxThisTick
         }
+        if (burnTime < 0) {
```
Comment (elifoster) [inline src/main/java/eiteam/esteemedinnovation/boiler/TileEntityBoiler.java:219]: Why is this added?

File: src/main/java/eiteam/esteemedinnovation/transport/fluid/screw/TileEntityPump.java [hunk 7]
```diff
@@ -127,25 +127,27 @@ public void safeUpdate() {
             offsetPos = getOffsetPos(outputDir);
             if (myTank.getFluidAmount() > 0 && progress == 100) {
                 TileEntity tile = world.getTileEntity(offsetPos);
```
Comment (elifoster) [inline src/main/java/eiteam/esteemedinnovation/transport/fluid/screw/TileEntityPump.java:129]: These changes don't appear to be relevant to the Boiler?

**Review by elifoster — CHANGES_REQUESTED** [review]
Have a couple suggestions and questions clarifying you reasons for some changes. Looks good generally

File: src/main/java/eiteam/esteemedinnovation/boiler/TileEntityBoiler.java (hunk 4 above)
Comment (ExpensiveKoala) [reply src/main/java/eiteam/esteemedinnovation/boiler/TileEntityBoiler.java:87]: 99% of the time, it should get caught by the
```java
  int burnTime = ForgeEventFactory.getItemBurnTime(stack);
  if (burnTime >= 0) {
      return burnTime;
  }
```
which replaces the `GameRegistry.getFuelStack(stack)`.
The other if statements really shouldn't be needed, but was included in the style that the furnace does it.

File: src/main/java/eiteam/esteemedinnovation/boiler/TileEntityBoiler.java (hunk 5 above)
Comment (ExpensiveKoala) [reply src/main/java/eiteam/esteemedinnovation/boiler/TileEntityBoiler.java]: It would be empty if no item is placed in the slot, no?

File: src/main/java/eiteam/esteemedinnovation/boiler/TileEntityBoiler.java (hunk 6 above)
Comment (ExpensiveKoala) [reply src/main/java/eiteam/esteemedinnovation/boiler/TileEntityBoiler.java:219]: There was an issue where the burn time was somehow getting negative. This resolves that issue.

File: src/main/java/eiteam/esteemedinnovation/transport/fluid/screw/TileEntityPump.java (hunk 7 above)
Comment (ExpensiveKoala) [reply src/main/java/eiteam/esteemedinnovation/transport/fluid/screw/TileEntityPump.java:129]: These changes were to make the screw functional enough to test the boiler. I had to test if the boiler still properly accepted fluid input, but didn't have anything that worked. I chose to fix this instead of adding another mod into my dev environment.

File: src/main/java/eiteam/esteemedinnovation/boiler/ContainerBoiler.java (hunk 2 above)
Comment (ExpensiveKoala) [reply src/main/java/eiteam/esteemedinnovation/boiler/ContainerBoiler.java:116]: isUsableByPlayer is a method from IInventory. Since we are no longer using it, this method does not exist anymore.

File: src/main/java/eiteam/esteemedinnovation/boiler/ContainerBoiler.java (hunk 3 above)
Comment (ExpensiveKoala) [reply src/main/java/eiteam/esteemedinnovation/boiler/ContainerBoiler.java:114]: It isn't marked `@Nonnull` in the interface where it comes from. Also, we aren't using it.

File: src/main/java/eiteam/esteemedinnovation/boiler/TileEntityBoiler.java (hunk 5 above)
Comment (elifoster) [reply src/main/java/eiteam/esteemedinnovation/boiler/TileEntityBoiler.java]: Right, what I mean is that `FluidHelper.itemStackIsWaterContainer(ItemStack)` returns false if `ItemStack#isEmpty`, so there's no need to check it twice.

File: src/main/java/eiteam/esteemedinnovation/boiler/TileEntityBoiler.java (hunk 6 above)
Comment (elifoster) [reply src/main/java/eiteam/esteemedinnovation/boiler/TileEntityBoiler.java:219]: It gets set to -1 on line 219. To be honest I don't quite understand how the burn time stuff works.

File: src/main/java/eiteam/esteemedinnovation/transport/fluid/screw/TileEntityPump.java (hunk 7 above)
Comment (elifoster) [reply src/main/java/eiteam/esteemedinnovation/transport/fluid/screw/TileEntityPump.java:129]: Ah, that makes sense.

File: src/main/java/eiteam/esteemedinnovation/boiler/TileEntityBoiler.java (hunk 6 above)
Comment (ExpensiveKoala) [reply src/main/java/eiteam/esteemedinnovation/boiler/TileEntityBoiler.java:219]: That's a -= not = -

File: src/main/java/eiteam/esteemedinnovation/boiler/TileEntityBoiler.java (hunk 5 above)
Comment (ExpensiveKoala) [reply src/main/java/eiteam/esteemedinnovation/boiler/TileEntityBoiler.java]: I guess yeah.

File: src/main/java/eiteam/esteemedinnovation/boiler/TileEntityBoiler.java (hunk 6 above)
Comment (elifoster) [reply src/main/java/eiteam/esteemedinnovation/boiler/TileEntityBoiler.java:219]: Oh you're right. I misread that.
